components/header-bar/src/user-icon/image-icon.js

```diff
@@ -1,22 +1,37 @@
+import { useConfig } from '@dhis2/app-runtime'
 import PropTypes from 'prop-types'
 import React from 'react'
 
-export const ImageIcon = ({ src, dataTest }) => (
-    <>
-        <img src={src} alt="user avatar" data-test={dataTest} />
+const useAvatarImgSrc = (avatarId) => {
+    const { baseUrl } = useConfig()
 
-        <style jsx>{`
-            img {
-                width: 100%;
-                height: 100%;
-                border-radius: 50%;
-                object-fit: cover;
-            }
-        `}</style>
-    </>
-)
+    return avatarId
+        ? [baseUrl, 'api/fileResources', avatarId, 'data']
+              .filter((part) => !!part)
+              .map((part) => part.replace(/^\/+|\/+$/g, ''))
+              .join('/')
+        : null
```

Comment:
I don't remember the reason for that complicated way to build the URL, to me it seems that something like this should work fine.

```suggestion
        ? new URL(`api/fileResources/${avatarId}/data`, baseUrl).href
        : null
```

Reply:
> I don't remember the reason for that complicated way to build the URL

The `filter + map + join` would be required for [a more generic `joinPath` function](https://github.com/dhis2/ui/blob/d7e1693bf6c75fa68e982793d95bc8bf21cfe2fb/components/header-bar/src/join-path.js) (which we use in the header bar for generating app URLs in the app drawer).

For the `UserAvatar` use case I agree that your solution will work fine.